### PR TITLE
Agents

### PR DESCRIPTION
**Pull request by gdiazlo**
Clean up agents before its transformation.

### CODE REVIEW

File: gossip/monitor/monitor.go [hunk 1]
```diff
@@ -55,84 +81,71 @@ func DefaultConfig() *Config {
...
-
 	monitor.executionTicker = time.NewTicker(conf.TaskExecutionInterval)
 	go monitor.runTaskDispatcher()
 
 	return &monitor, nil
 }
 
-type Task interface {
-	Do()
+func (m Monitor) RegisterMetrics(r *prometheus.Registry) {
```
Comment (iknite) [inline gossip/monitor/monitor.go]: If you're going to change the way we register metrics IMHO, you should change every occurrence of this code, or at least make an Issue, tech debt post-it or whatever to end this task.

File: gossip/monitor/monitor.go [hunk 2]
```diff
@@ -195,38 +201,38 @@ func (m Monitor) dispatchTasks() {
...
-		log.Infof("Error saving batch in alertStore (task re-enqueued): %v", err)
+		log.Infof("Monitor had an error saving batch in alertStore (task re-enqueued): %v", err)
 		q.taskCh <- q
 		return
 	}
 	defer resp.Body.Close()
 	_, err = io.Copy(ioutil.Discard, resp.Body)
 	if err != nil {
-		log.Infof("Error getting response from alertStore saving a batch: %v", err)
+		log.Infof("MOnitor had an error from alertStore saving a batch: %v", err)
```
Comment (iknite) [inline gossip/monitor/monitor.go]: *MO*nitor (typo)

File: tests/e2e/test_service.go [hunk 3]
```diff
@@ -108,43 +118,42 @@ func (a *alertStore) GetAll() []string {
...
 type snapStore struct {
-	segments []Segment
-	sync.Mutex
+	data *freecache.Cache
 }
 
 func newSnapStore() *snapStore {
-	return &snapStore{segments: append([]Segment{}, Segment{})}
+	return &snapStore{
+		data: freecache.NewCache(2 << 30),
```
Comment (iknite) [inline tests/e2e/test_service.go]: Maybe a small comment of this "magic number"? (why it's 2<<30 at least)

File: gossip/monitor/monitor.go (hunk 1 above)
Comment (gdiazlo) [reply gossip/monitor/monitor.go]: It is changed in all processors.

File: tests/e2e/test_service.go (hunk 3 above)
Comment (gdiazlo) [reply tests/e2e/test_service.go]: That's a gigabyte of cache. Its not a magic number. Seems good enough for our testing environment. No other particular reason.

**Pull request merged**